### ISSUE TITLE
[0.9.2 staging-1884] Should delete possibility to 'Become Resident' if titles or demographics were removed.

### ISSUE THREAD

**Issue by SlayksWood** [StrangeLoopGames/EcoIssues#19845]
Step to reproduce:
- Make title and add user to it:
![image](https://user-images.githubusercontent.com/45708377/103079240-861f7280-45e4-11eb-991e-b9c4adb7580b.png)
- invite this title to residency:
![image](https://user-images.githubusercontent.com/45708377/103079329-b36c2080-45e4-11eb-93f1-9a734ab676a8.png)
- delete title:
![image](https://user-images.githubusercontent.com/45708377/103079363-c121a600-45e4-11eb-8a67-ff51c10207d1.png)
- open property window:
![image](https://user-images.githubusercontent.com/45708377/103079394-d0085880-45e4-11eb-9801-91e484e73ec7.png)
- don't delete it and User (which you add to this title) try to become a resident, he can!:
![image](https://user-images.githubusercontent.com/45708377/103079447-e9110980-45e4-11eb-9320-69b1f033c88b.png)

So first, User should not become a resident, second, delete removed title from UI.

**Comment by SlayksWood** [comment]
Checked in 0.9.3 staging-1928 still an issue:
![image](https://user-images.githubusercontent.com/45708377/107733807-f4db9b00-6d0c-11eb-8dcf-5b2a283dc769.png)

**Comment by Daniel-Magulado** [comment]
Tested on 0.9.3 Staging-1951.

I am still able to reproduce the issue.

Video Attached:
https://images.zenhubusercontent.com/47210472/14122d55-652c-4f65-a12c-65a4a97dd863/untitled.mp4

Note:
On my repro:
1. Create title.
2. Assign a player as the titleholder of the title you created in step 1. (I assigned myself)
3. Invite title to deed to become resident.
4. Delete the title you created in step 1. Delete in Registrar.
5. Give the ownership of the deed/property to another player. I created test users here via /test makeusers. I gave the ownership of the deed to Test Client 1. Then close Deed UI.
6. Open the deed UI that you gave the ownership to Test Client 1. Click Become Resident.
Result:
The Player still can become a resident of the said deed.

Expected result
1. If the title is invited to become a resident and the player that holds that title did not set itself or accepted to become a resident, if the title is removed, the invite should be deleted as well or consumed.
2. Deleted titles should be deleted on every UI where it was displayed or present.

What happening is that inviting the title gives the titleholder a ticket to become a resident regardless of the title condition if it still exists or removed. If the title is deleted, the titleholder should lose its ticket to become a resident.
